testsuites/benchmark/src/main/java/org/eclipse/rdf4j/benchmark/rio/util/DataSetGenerator.java

```diff
@@ -0,0 +1,430 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.benchmark.rio.util;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Objects;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.RDFWriter;
+import org.eclipse.rdf4j.rio.Rio;
+import org.eclipse.rdf4j.rio.binary.BinaryRDFWriter;
+
+/**
+ * Synthetic dataset generator.
+ *
+ * @author Tomas Kovachev t.kovachev1996@gmail.com
+ */
+public class DataSetGenerator {
+	public static final String NAMESPACE = "http://example.com/gererator/";
+	private static final String WORD_DIR = "generated-data/words.txt";
+	private static final String DIR = "src/main/resources/generated-data/";
+	private static final int TOTAL_PREDS = 500;
+	private static final int AVG_GRAPH_SIZE = 500;
+	private final ValueFactory factory;
+	private final StringBuilder valueBuilder;
+	private final AtomicInteger contextCount = new AtomicInteger(0);
+	private final AtomicInteger maxCurrentContextCount = new AtomicInteger(0);
+	private ArrayList<String> dictionary;
+	private ArrayList<IRI> predicateIRIs;
+	private ArrayList<IRI> contextIRIs;
+	private Random randomResource;
+	private Random randomProbability;
+	private BNode prevBNode;
+	private IRI currentContext;
+	private int totalWords;
+	private int avgWordSize;
+
+	public DataSetGenerator() {
+		this.factory = SimpleValueFactory.getInstance();
+		this.valueBuilder = new StringBuilder();
+		this.randomResource = new Random(64);
+		this.randomProbability = new Random(64);
+		this.avgWordSize = 0;
+		fetchWords();
+		generatePredicateList();
+	}
+
+	public File generateStatementsFile(RDFFormat format, String fileName, int percentBNodes, int percentLiterals,
+			int textMinLength, int textMaxLength, int totalStatements, boolean isTextOnly,
+			boolean withContext) throws IOException {
```

Comment:
This is very nice! Thank you :)

Btw. If you want to parse a real world file I regularly use the two following files:
 - https://github.com/eclipse/rdf4j/blob/main/core/sail/memory/src/test/resources/benchmarkFiles/datagovbe-valid.ttl (remember to also add https://github.com/eclipse/rdf4j/blob/main/core/sail/memory/src/test/resources/benchmarkFiles/LICENSE)
 -  https://github.com/eclipse/rdf4j/blob/75b97f6106e6a73776d7f0f663f18a3edfb0a4ae/compliance/model/src/test/resources/benchmarkFiles/bsbm-100.ttl

You don't have to use these files though. It was just to let you know what we currently have.